Option to follow symbolic links in Full backup

maselvaraj opened the pull request:
Adds a checkbox that enables "follow symlinks" in "Full backup" which fixes the issue https://github.com/emanuelez/hudson-periodicbackup/issues/1

If Jenkins home is a symlink to another directory, then backup would fail since code ignores symlinks. This PR adds an option that a user can select to follow symlinks. If this check box is enabled, then full backup will work even if Jenkins home is a symlink.

Related PR that disabled symlink : https://github.com/jenkinsci/periodicbackup-plugin/pull/9 . I think it's better to give the choice to the user to follow symlinks or not. While the default is still not to follow symlinks.

maselvaraj commented:
@oleg-nenashev Can you please review this PR / add someone who can help with the review 
Thanks in advance 🙇 !

maselvaraj commented:
@emanuelez @johnou Can you please review this PR ? TIA 🙏

oleg-nenashev commented:
I stepped down as a maintainer long long ago. I have no bandwidth for this plugin anymore due to other roles in the Jenkins project, and anyone interested in the plugin is welcome to adopt it. Sorry for such response. https://jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/

maselvaraj commented:
Thanks @oleg-nenashev for your reply! I'll reach out in the mailing list for adopting the plugin.

johnou commented on src/main/java/org/jenkinsci/plugins/periodicbackup/FullBackup.java:
```diff
@@ -52,14 +52,15 @@
     @DataBoundConstructor
-	public FullBackup(@CheckForNull String excludesString) {
-		this(excludesString, Jenkins.getActiveInstance().getRootDir());
+	public FullBackup(@CheckForNull String excludesString, boolean followSymbolicLinks) {
```
If I am not mistaken the old constructor annotated with DataBoundConstructor needs to remain or old configurations will not be loaded?

maselvaraj replied:
I tested with the old configuration and it is working fine.

johnou commented on src/main/java/org/jenkinsci/plugins/periodicbackup/FullBackup.java:
```diff
@@ -68,9 +69,10 @@ public FullBackup(@CheckForNull String excludesString) {
      * @param baseDir Base directory
      */
-    FullBackup(@CheckForNull String excludesString, @Nonnull File baseDir) {
+    FullBackup(@CheckForNull String excludesString, @CheckForNull boolean followSymbolicLinks, @Nonnull File baseDir) {
```
CheckForNull redundant, primitives cannot be null.

maselvaraj replied:
removed it!

johnou commented on src/main/java/org/jenkinsci/plugins/periodicbackup/FullBackup.java:
```diff
@@ -121,6 +123,11 @@ public String getExcludesString() {
         return excludesString;
     }
 
+    @CheckForNull
```
primitive, not required.

maselvaraj replied:
removed it!

maselvaraj commented:
Thanks much @johnou for reviewing! I addressed your comments and I've tested the backward compatibility with old config and it is working fine. I don't see any issues.

johnou commented:
Thanks @maselvaraj!

oleg-nenashev commented:
@johnou Hi. Do you take ownership of the plugin back? I am reworking the adoption process, and I wonder whether I should mark it for adoption in GitHub

johnou commented:
@oleg-nenashev sure, it's pretty low traffic anyway. My sonatype credentials should still be valid, right?

maselvaraj commented:
@johnou Would you be able to cut a release for the plugin 🙇 ?

johnou commented:
@maselvaraj currently waiting for permission so I can push releases again.

johnou commented:
@maselvaraj released.

maselvaraj commented:
Awesome ! Thank you  @johnou 🙇 !

johnou commented:
Thanks for the contribution.
